### DISCUSSION TITLE
Dynamic Config Contexts

### DISCUSSION THREAD

**Discussion by tophe75** [netbox-community/netbox#10427] [Q&A] [answered]
Hi.

I am trying to render config contexts dynamically from device fields

`{
    "System": {
        "hostname": "{{ obj.name }}",
        "mgmt_intf": {
            "ip": "{{ obj.primary_ip4.address.ip }}",
            "mode": "static",
            "subnet_mask": "24"
        }
    }
}`

Would this be possible or can the fields not be rendered within the config contexts?

**Comment by candlerb** [comment, marked as answer]
AFAIK, config contexts are static JSON data.  If you want to perform expansion on the values you receive, this is something you'd have to do externally.
